### PR TITLE
Add ValidationRule

### PR DESCRIPTION
**Pull request by mahdimsr**
add a validation rule to validate the token in the request

### CODE REVIEW

File: src/ValidationRule.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Fouladgar\OTP;
+
+use Fouladgar\OTP\Tests\Models\OTPNotifiableUser;
+use Illuminate\Support\Facades\Cache;
```
Comment (mohammad-fouladgar) [inline src/ValidationRule.php:6]: Please remove unused namespaces;

File: src/ValidationRule.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Fouladgar\OTP;
+
+use Fouladgar\OTP\Tests\Models\OTPNotifiableUser;
```
Comment (mohammad-fouladgar) [inline src/ValidationRule.php:5]: You should use the `Fouladgar\OTP\Contracts\OTPNotifiable` interface.

**Comment by mohammad-fouladgar** [comment]
@mahdimsr  Thanks for your PR. 
I'd like to know what your idea is about this PR? This validation rule just returns true or false. In case we need the user validated by mobile and token to login user.

For example, you are using sanctum:
```php
   $user  = OTP()->validate('+989389599530', 'token_123');
   $token = $user->createToken('ios')->plainTextToken;

   auth()->login($user);
```

**Comment by mahdimsr** [comment]
> @mahdimsr Thanks for your PR. I'd like to know what your idea is about this PR? This validation rule just returns true or false. In case we need the user validated by mobile and token to login user.
> 
> For example, you are using sanctum:
> 
> ```
>    $user  = OTP()->validate('+989389599530', 'token_123');
>    $token = $user->createToken('ios')->plainTextToken;
> 
>    auth()->login($user);
> ```

the validation you used in your example is enough to validate and check. but what if the developer wants to return some error message if the token isn't valid? two options are to choose from:

- use the validate function like your example and return an error message
in this way, you have to return the error badge message manually
- use request validate rule

**Comment by mohammad-fouladgar** [comment]
@mahdimsr I agree with you. But there is a problem. When you call the `validate` method, this method [revokes](https://github.com/mohammad-fouladgar/laravel-otp/blob/05decb14bf8dd52d4a415bcb52d0a2c763d103e6/src/OTPBroker.php#L64) the token, therefore, we lose the token and we can not find the user, right?
IMO, we can have a method something like that in the `Fouladgar\OTP\OTPBroker` class:
```php
...
public function isValidToken(OTPNotifiable $user, string $token): bool
{
    return $this->tokenExists($user,$token);
}
...
```
And in our rule, use this method, not the `validate` method.

**Comment by mahdimsr** [comment]
> @mahdimsr I agree with you. But there is a problem. When you call the `validate` method, this method [revokes](https://github.com/mohammad-fouladgar/laravel-otp/blob/05decb14bf8dd52d4a415bcb52d0a2c763d103e6/src/OTPBroker.php#L64) the token, therefore, we lose the token and we can not find the user, right? IMO, we can have a method something like that in the `Fouladgar\OTP\OTPBroker` class:
> 
> ```
> ...
> public function isValidToken(OTPNotifiable $user, string $token): bool
> {
>     return $this->tokenExists($user,$token);
> }
> ...
> ```
> 
> And in our rule, use this method, not the `validate` method.

oh, I didn't recognize that. you're right :). I will fix it. thanks for the comment

**Comment by mohammad-fouladgar** [comment]
Dear @mahdimsr
 Because this issue seems to be inactive for quite some time now. I've closed this PR. Please feel free to reopen it.

**Pull request closed without merging**